api/resources/annotations.py

```diff
@@ -0,0 +1,42 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Exists, IntegerField, OuterRef, Subquery
+
+from osf.models import OutcomeArtifact, Registration
+from osf.utils.outcomes import ArtifactTypes
+
+
+def _make_badge_subquery(artifact_type):
+    '''Make the subquery for a specific Open Practice Badge.
+
+    Requires `primary_outcome_id` annotation be present. As such, this should only be called
+    from make_open_practice_badge_annotations.
+    '''
+    return Exists(
+        OutcomeArtifact.objects.filter(
+            outcome_id=OuterRef('primary_outcome_id'),
+            finalized=True,
+            deleted__isnull=True,
+            artifact_type=artifact_type,
+        ),
+    )
+
+
+def make_open_practice_badge_annotations():
+    '''Builds all of the annotation Subqueries for OpenPractice badges. '''
+    #TODO: Accept CT as a param
```

Comment:
Minor: Is this `TODO` necessary? Should it be implemented now?

Reply:
Honestly, I'd like to add something like

```
@functools.lru_cache
@classmethod
def get_content_type(cls):
  return ContentType.objects.get_for_model(cls)
```

to `BaseMode`l to make this easier everywhere we need it, but that felt more contentious than what I wanted to do in this PR :)

Reply:
`BaseModel` has existed longer than we've used `ContentType`; it's basically a ported adapter from the pre-Django days and has received little attention since (as evidenced by the fact that it still has a `migration_page_size`). 

I wouldn't consider that addition to be particularly contentious, but it might be scope creep.

Reply:
Leaving this as a TODO since it's unclear if we will ever need to support non-Registrations as Primary Resources